spi/src/main/java/io/quarkiverse/kafkastreamsprocessor/spi/KStreamsProcessorConfigGenerator.java

```diff
@@ -0,0 +1,116 @@
+import io.quarkiverse.kafkastreamsprocessor.spi.properties.RetryConfig;
+import io.quarkiverse.kafkastreamsprocessor.spi.properties.SinkConfig;
+import io.quarkiverse.kafkastreamsprocessor.spi.properties.SourceConfig;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+
+class KStreamsProcessorConfigGenerator {
+
+    private static final String SOURCE_PREFIX = "kafkastreamsprocessor.input.sources.";
+
+    private static final String SOURCE_SUFFIX = ".topics";
+
+    private static final String SINK_PREFIX = "kafkastreamsprocessor.output.sinks.";
+
+    private static final String SINK_SUFFIX = ".topic";
+
+    static KStreamsProcessorConfig buildConfig(Config config) {
+        InputConfigImpl input = new InputConfigImpl(config.getOptionalValue("kafkastreamsprocessor.input.topic", String.class),
+                config.getOptionalValues("kafkastreamsprocessor.input.topics", String.class));
+        fillSources(config, input);
+        OutputConfigImpl output = new OutputConfigImpl(
+                config.getOptionalValue("kafkastreamsprocessor.output.topic", String.class));
+        fillSinks(config, output);
+        return new KStreamsProcessorConfigImpl(input, output);
+    }
+
+    private static void fillSources(Config config, InputConfigImpl input) {
+        for (String property : config.getPropertyNames()) {
+            if (property.startsWith(SOURCE_PREFIX) && property.endsWith(SOURCE_SUFFIX)) {
+                String sourceName = property.substring(SOURCE_PREFIX.length(), property.length() - SOURCE_SUFFIX.length());
+                String topics = config.getValue(property, String.class);
+                SourceConfig sourceConfig = new SourceConfigImpl(List.of(topics.split(",")));
+                if (sourceName.contains(".")) {
+                    throw new IllegalStateException("Parsed source name has a dot: " + sourceName);
+                }
+                input.sources().put(sourceName, sourceConfig);
+            }
+        }
+    }
+
+    private static void fillSinks(Config config, OutputConfigImpl output) {
+        for (String property : config.getPropertyNames()) {
+            if (property.startsWith(SINK_PREFIX) && property.endsWith(SINK_SUFFIX)) {
+                String sinkName = property.substring(SINK_PREFIX.length(), property.length() - SINK_SUFFIX.length());
+                String topic = config.getValue(property, String.class);
+                SinkConfig sinkConfig = new SinkConfigImpl(topic);
+                if (sinkName.contains(".")) {
+                    throw new IllegalStateException("Parsed sink name has a dot: " + sinkName);
+                }
+                output.sinks().put(sinkName, sinkConfig);
+            }
+        }
+    }
+
+    @RequiredArgsConstructor
+    @Getter
+    @Accessors(fluent = true)
+    private static class KStreamsProcessorConfigImpl implements KStreamsProcessorConfig {
```

Comment:
and do we need classes to be annotated with @ConfigMapping ?

Reply:
The idea of the PR is to make the fix the less intrusive possible.
To keep a ConfigMapping interface because it makes the Configuration properties definitions much cleaner.
And a capability to get the sink to topic and source to topics mappings with a pure Config object.

Readme enhanced.